gcc/d/d-objfile.cc

```diff
@@ -1010,10 +1008,15 @@ build_moduleinfo_symbol(Module *m)
 
   flags |= MIname;
 
+  tree msym = get_moduleinfo_decl (m);
+  TREE_TYPE (msym) = layout_moduleinfo_fields (m, TREE_TYPE (msym));
```

Comment:
This call was moved here, as there is a dependency problem - `get_moduleinfo_decl` may be called to get the forward-referenced module decl.